js/router.js

```diff
@@ -117,6 +139,59 @@ module.exports = Backbone.Router.extend({
     $('#obContainer')[0].scrollTop = 0;
   },
 
+  launchPageConnectModal: function(config) {
+    if (!(
+        config &&
+        config.promise &&
+        typeof config.promise.then === 'function'
+      )) {
+      throw new Error('At a minimum, the config must contain a config.promise.');
+    }
+
+    var defaults = {
+      connectText: 'Connecting...',
+      failedText: 'Unable to Connect.'
+    };
+
+    config = __.extend({}, defaults, config);
+
+    this.pageConnectModal && this.pageConnectModal.remove();
+    this.pageConnectModal = new PageConnectModal({
+      initialState: {
+        statusText: config.connectText,
+        tooltip: config.connectTooltip
+      }
+    }).render().open();    
+
+    this.pageConnectModal.on('back', () => {
+      history.back();
+    });
+
+    this.pageConnectModal.on('retry', () => {
+      // recall the same route
+      var route = location.hash;
+
+      this.navigate('blah-blah', { replace: true });
+      this.navigate(route, { replace: true, trigger: true });
+    });
+
+    this.pageConnectModal.on('cancel', () => {
+      typeof config.promise.cancel === 'function' && config.promise.cancel();
+      history.back();
```

Comment:
Cancel should deactivate the button when it's clicked, it can take a few seconds for the loading screen to disappear, and you can click the button multiple times leading to history going back several places.